server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxChangesMethod.scala

```diff
@@ -64,7 +64,12 @@ class MailboxChangesMethod @Inject()(mailboxSerializer: MailboxSerializer,
         methodCallId = invocation.invocation.methodCallId
       ), processingContext = invocation.processingContext))
     } else {
-      SMono.fromPublisher(mailboxChangeRepository.getSinceState(JavaAccountId.fromUsername(mailboxSession.getUser), JavaState.of(request.sinceState.value), request.maxChanged.toJava, capabilities.contains(CapabilityIdentifier.JAMES_DELEGATED_CHANGE)))
+      SMono.fromPublisher(
+        if (capabilities.contains(CapabilityIdentifier.JAMES_SHARES)) {
+          mailboxChangeRepository.getSinceStateWithDelegation(JavaAccountId.fromUsername(mailboxSession.getUser), JavaState.of(request.sinceState.value), request.maxChanged.toJava)
+        } else {
+          mailboxChangeRepository.getSinceState(JavaAccountId.fromUsername(mailboxSession.getUser), JavaState.of(request.sinceState.value), request.maxChanged.toJava)
```

Comment:
`JavaAccountId.fromUsername(mailboxSession.getUser)` & `JavaState.of(request.sinceState.value)`

could be vals allocated out of the if?